blog/2023-01-17-mongodb-monitoring.md

```diff
@@ -0,0 +1,297 @@
+ 
+![image](https://user-images.githubusercontent.com/24860547/212876243-f96fae2a-b5c1-4ecc-b425-8a9dbd38424b.png)
+
+### Memory
+ - **Page Faults**: The number of page faults encountered by the MongoDB server per second. 
+
+Page faults occur when the MongoDB server needs to access parts of data that are not currently in the memory. As the dataset size increases, more page faults will occur unless the working set of data can be kept in memory. Monitoring page faults helps to identify when MongoDB is not able to keep the working set in memory. If the page faults are too high, it may indicate that the dataset size is too large for the available memory, and the server should be upgraded or the data should be partitioned.
+
+In addition, monitoring page faults can help identify if the dataset is not being accessed in an optimal way. If the page faults are too high, it may indicate that the queries are not using the correct indexes or that a sufficient number of indexes are not used to cover the query.
+
+Page faults can also help identify if the MongoDB server is being overloaded. If the page faults are too high and cannot be attributed to dataset size, indexing, or query optimization, it may indicate that the server is not configured correctly or that there are too many connections for the server to handle.
+
+ - **TCMalloc**: The usage of the TCMalloc allocator by the MongoDB server. The usage in bytes for each of the following is monitored: 
+	- pageheap_free
+	- pageheap_unmapped 
+	- total_threaded_cache
+	- free
+	- pageheap_committed
+	- pageheap_total_commit
+	- pageheap_decommit
+	- pageheap_reserve
+	
+
+TCMalloc is a thread-local storage allocator that is used by MongoDB to allocate memory for various operations. It is important to monitor the usage of TCMalloc as it can help prevent memory-related issues and performance bottlenecks. By monitoring the usage of TCMalloc, you can identify when more memory is needed, when the memory is not being released back to the operating system, and when memory fragmentation is occurring.
+
+For example, if the pageheap_free metric is consistently low and pageheap_unmapped is consistently high, it could indicate that memory is not being released back to the operating system, resulting in memory fragmentation. This can cause performance issues and lead to out-of-memory errors.
+
+Monitoring the total_threaded_cache metric can help identify when more memory is needed, as it shows the total amount of memory allocated by TCMalloc in bytes. If this metric is consistently high, it could indicate that more memory is needed to handle the workload.
+
+Monitoring the free metric can also help indicate when more memory is needed, as it shows the amount of free memory available to TCMalloc in bytes. If this metric is consistently low, it could indicate that more memory is needed to handle the workload. 
```

Comment:
Is this really the description coming from the chart info? May be too long for a Netdata user who already knows the subject.

Reply:
No, I ran it through another round of GPT to make it more verbose. Definitely too long for putting inside the app info description, but fine for a blog I think.